### PR TITLE
Move abandoned cart to free [MAILPOET-5576]

### PR DESCRIPTION
**Pull request by websupporter**
## Description
Moves the abandoned cart trigger to the free plugin

## Code review notes
A test for the new trigger was missing. I added one.

## Linked PRs

https://github.com/mailpoet/mailpoet-premium/pull/783

## Linked tickets

[MAILPOET-5576]

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5576]: https://mailpoet.atlassian.net/browse/MAILPOET-5576?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/tests/integration/Automation/Integrations/WooCommerce/Triggers/AbandonedCartTriggerTest.php [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    $productId = wp_insert_post($productData);
+    $this->assertIsInt($productId);
+    update_post_meta($productId, '_price', $price);
+    return $productId;
+  }
+
+  public function _after() {
+    wp_set_current_user(0);
+    $this->tasksRepository->truncate();
+    parent::_after(); // TODO: Change the autogenerated stub
```
Comment (rodrigoprimo) [inline mailpoet/tests/integration/Automation/Integrations/WooCommerce/Triggers/AbandonedCartTriggerTest.php]: I guess this TODO is not necessary anymore

**Review by rodrigoprimo — COMMENTED** [review]
Looks good to me. I just left one comment with a tiny nitpick.

File: mailpoet/tests/integration/Automation/Integrations/WooCommerce/Triggers/AbandonedCartTriggerTest.php (hunk 1 above)
Comment (websupporter) [reply mailpoet/tests/integration/Automation/Integrations/WooCommerce/Triggers/AbandonedCartTriggerTest.php]: Fixed in https://github.com/mailpoet/mailpoet/pull/5149/commits/e89b970adf97d903bbccb96738105eff5074cf17. Assigning this to QA

**Review by rodrigoprimo — APPROVED** [review]

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**